tests/ui/methods/method-self-arg-trait.rs

```diff
@@ -1,7 +1,9 @@
 //@ run-pass
 // Test method calls with self as an argument
 
-static mut COUNT: u64 = 1;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+static COUNT: AtomicU32 = AtomicU32::new(1);
```

Comment:
@obeis: Looking at this failing test, I'd suggest dropping the atomics entirely here, and to just revert the test to how it was originally, except to replace e.g.  `COUNT *= 7` with `COUNT = COUNT * 7`.  This would avoid taking the reference (via `MulAssign`) with a minimum of other changes.